### PR TITLE
Fix problem where , and " in stub outputs are ignored

### PR DESCRIPTION
**Pull request by bennyhat**

### CODE REVIEW

File: bin/bash_stub.sh [hunk 1]
```diff
@@ -55,8 +55,12 @@ function send-to-server {
...
 }
 
-function extract-properties {
-  echo -n "${1}" | sed -En "s/^\"${2}\":\"?([^,\"]*)\"?,?$/\1/gp"
+function extract-number-properties {
+  echo -n "${1}" | sed -En "s/^\"${2}\":([0-9]+),?$/\1/gp"
+}
+
+function extract-string-properties {
+  echo -n "${1}" | sed -En "s/^\"${2}\":\"(.*)\",?$/\1/gp"
```
Comment (mdurban) [inline bin/bash_stub.sh]: It doesn't really matter much, but these two regexes are almost identical. I wonder if we should extract the core regex that they share

**Review by mdurban — APPROVED** [review]

File: bin/bash_stub.sh (hunk 1 above)
Comment (bennyhat) [reply bin/bash_stub.sh]: I agree. I think the biggest problem with that is the weird way bash deals with the variables. That is, the common part includes interpolation. Could be easy, could be hard?!

Comment (bennyhat) [reply bin/bash_stub.sh]: I'm waiting to hear back from our QA team :) If they think this is good, I'll merge and release just to get this nasty but out of there. Such a big miss :(

Comment (mdurban) [reply bin/bash_stub.sh]: It's nearly impossible to not miss something when dealing with regexes...

**Pull request merged**